lisp/tree-sitter-cli.el

```diff
@@ -16,21 +16,51 @@
 (eval-when-compile
   (require 'subr-x))
 
-(defun tree-sitter-cli-directory ()
-  "Return tree-sitter CLI's directory, including the ending separator.
-This is the directory where the CLI tool keeps compiled lang definitions, among
-other data."
+(defvar tree-sitter-version (nth 1 (split-string (shell-command-to-string "tree-sitter -V")))
+  "Return tree-sitter CLI version.")
+
+(defvar tree-sitter-version-alist
+  (let ((versions (split-string tree-sitter-version "\\.")))
+    (list
+     (cons 'major (string-to-number (nth 0 versions)))
+     (cons 'minor (string-to-number (nth 1 versions)))
+     (cons 'patch (string-to-number (nth 2 versions)))))
+  "Return alist of tree-sitter CLI semantic versioning.")
+
+(defun tree-sitter-cli-config-directory ()
+  "Return tree-sitter CLI's config directory, including the ending separator.
+This is the directory where the CLI stores the configuration file."
   (file-name-as-directory
    (expand-file-name
     ;; https://github.com/tree-sitter/tree-sitter/blob/1bad6dc/cli/src/config.rs#L20
     (if-let ((dir (getenv "TREE_SITTER_DIR")))
         dir
       "~/.tree-sitter"))))
 
-(defun tree-sitter-cli-bin-directory ()
+(defun tree-sitter-cli-cache-directory ()
+  "Return tree-sitter CLI's cache directory, including the ending separator.
+This is the directory where the CLI tool keeps compiled lang definitions."
+  (file-name-as-directory
+   ;; https://github.com/tree-sitter/tree-sitter/blob/master/cli/loader/src/lib.rs#L110-L115
+   (expand-file-name "tree-sitter"
+                     (cond
+                      ((eq system-type 'gnu/linux)
+                       (let ((env (getenv "XDG_CACHE_HOME")))
+                         (if (or (null env) (not (file-name-absolute-p env)))
+                             (expand-file-name "~/.cache")
+                           env)))
+                      ((eq system-type 'darwin)
+                       "~/Library/Caches")
+                      ((memq system-type '(cygwin windows-nt ms-dos))
+                       "~/AppData/Local")))))
+
+(defun tree-sitter-cli-lib-directory ()
   "Return the directory used by tree-sitter CLI to store compiled grammars."
   (file-name-as-directory
-   (concat (tree-sitter-cli-directory) "bin")))
+   (if (and (equal (alist-get 'major tree-sitter-version-alist) 0)
+            (< (alist-get 'minor tree-sitter-version-alist) 20))
+       (expand-file-name "bin" (tree-sitter-cli-config-directory))
+     (expand-file-name "lib" (tree-sitter-cli-cache-directory)))))
```

Comment:
```suggestion
   (if (version< "0.20" tree-sitter-version)
       (expand-file-name "bin" (tree-sitter-cli-config-directory))
     (expand-file-name "lib" (tree-sitter-cli-cache-directory)))))
```

Reply:
I wasnt aware of those functions. Thanks!